### ISSUE TITLE
Dense feature extraction

### ISSUE THREAD

**Issue by Davidyao99** [facebookresearch/dinov2#300]
Great Work!

I am hoping to use dinov2 to get pixel level dense feature extraction. I believe I should be doing this using the tutorial [here](https://github.com/facebookresearch/dinov2/issues/153)

I realize that the images are automatically resized to be 256x256. I am thinking of bilinear interpolation to retrieve a dense pixel level feature extraction. However, I was wondering if it will be better to not resize as much so that there are less degradation due to the interpolation. Should I try to prevent the image reshaping in the postprocessing by changing the AutoImageProcessor parameters? Any suggestions as to how I can get the best dense features without losing detail from interpolation? Thank you!!

**Comment by qasfb** [comment]
If you want higher-resolution maps, I think you can just feed a larger image to the model; I'm not personally familiar with AutoImageProcessor but I imagine there should be some parameters allowing it.

Otherwise you can use [this function](https://github.com/facebookresearch/dinov2/blob/main/dinov2/data/transforms.py#L77-L91), by adjusting the resize/crop kwargs, then feeding the output to the model to collect higher-resolution maps.

**Comment by Davidyao99** [comment]
@qasfb I see, that makes a lot of sense. Do you know if this will affect model results? I suppose the model was trained on cropped square images. If I want to extract dense features for non-cropped rectangular frames, I am thinking this might not be ideal? Just curious if anyone has any thoughts on this!!

**Comment by qasfb** [comment]
the model is expected to behave well up to at lease resolution 518x518. Rectangular might be a bit more touchy depending on the aspect ratio but you should give it a try, or maybe try a windowed approach. Only experiments will tell !

**Comment by 1921134176** [comment]
Sharing with you how I handled it. When extracting remote sensing image features, I found that when the input image is not larger than 2072x2072, the model still gets good features after direct input to the network and positional coded interpolation, and after further increasing the input, I feel that the effect becomes worse after feature clustering. I use the sliding window approach, the window size is 518, each slide 259, the overlap part of the average of the feature values, to extract the feature map of any size image. I refer to the mmsegmentation sliding window test code. Currently mmsegmentation supports dinov2, feature extraction is easy to implement.
